docs/data/routes/docs/nextjs/graphql/component/en.md

```diff
@@ -0,0 +1,42 @@
+---
+name: component
+routeTemplate: ./data/component-templates/article.yml
+title: Using GraphQL in component SSR/SSG
+---
+# Walkthrough: Using GraphQL in component SSR/SSG
+
+Next.js sample app provides you ability to use [component level data fetching](/docs/nextjs/data-fetching/component-level-data-fetching), read about it first.
+Consequently you can execute GraphQL queries on component level and use fetched data in component.
+
+> NOTE: In case if you are working in [Disconnected](/docs/techniques/working-disconnected/disconnected-overview) mode you can't use GraphQL queries, because disconnected server doesn't support GraphQL.
+
+Let's go through the steps to execute first GraphQL query in your component. We can use [GraphQL-ConnectedDemo](https://github.com/Sitecore/jss/blob/master/samples/nextjs/src/components/graphql/GraphQL-ConnectedDemo.tsx) example.
+
+1. Define `getStaticProps` or/and `getServerSideProps`. Next steps you should perform inside these functions.
+2. Create new GraphQL client. `GraphQLClientFactory` is using [apollo-client](https://www.apollographql.com/docs/react) library.
+```ts
+import GraphQLClientFactory from 'lib/GraphQLClientFactory';
+...
+const graphQLClient = GraphQLClientFactory();
+```
+3. Import needed GraphQL query. [graphql-let](https://github.com/piglovesyou/graphql-let) provides ability to import queries.
```

Comment:
Should we add details on how the query is expected to be created/defined? And do they need to `graphql:update` after creating it?

Should we add a hint on how to write/debug your query? (with GraphiQL)

Reply:
And what data endpoint this is using

Reply:
Added note: https://github.com/Sitecore/jss/pull/587/files#diff-f48c7acb9e727a477be3fca37355f49120f17b315fcb9c47421d6c37dff77bd6R18

Reply:
Added note about debug, etc: https://github.com/Sitecore/jss/pull/587/files#diff-f48c7acb9e727a477be3fca37355f49120f17b315fcb9c47421d6c37dff77bd6R24-R27